src/script/script.cpp

```diff
@@ -298,31 +297,38 @@ bool GetScriptOp(CScriptBase::const_iterator& pc, CScriptBase::const_iterator en
         }
         else if (opcode == OP_PUSHDATA1)
         {
-            if (end - pc < 1)
-                return false;
-            nSize = *pc++;
+            if (script.size() < 1) return false;
+            nSize = script[0];
+            script = script.subspan(1);
         }
         else if (opcode == OP_PUSHDATA2)
         {
-            if (end - pc < 2)
-                return false;
-            nSize = ReadLE16(&pc[0]);
-            pc += 2;
+            if (script.size() < 2) return false;
+            nSize = ReadLE16(script.data());
+            script = script.subspan(2);
         }
         else if (opcode == OP_PUSHDATA4)
         {
-            if (end - pc < 4)
-                return false;
-            nSize = ReadLE32(&pc[0]);
-            pc += 4;
+            if (script.size() < 4) return false;
+            nSize = ReadLE32(script.data());
+            script = script.subspan(4);
         }
-        if (end - pc < 0 || (unsigned int)(end - pc) < nSize)
-            return false;
-        if (pvchRet)
-            pvchRet->assign(pc, pc + nSize);
-        pc += nSize;
+        if (script.size() < nSize) return false;
+        if (data_out) *data_out = script.subspan(0, nSize);
+        script = script.subspan(nSize);
     }
 
     opcodeRet = static_cast<opcodetype>(opcode);
     return true;
 }
+
+bool GetScriptOp(CScriptBase::const_iterator& pc, CScriptBase::const_iterator end, opcodetype& opcodeRet, std::vector<unsigned char>* pvchRet)
+{
+    if (pc >= end) return false;
+    Span<const unsigned char> script(&*pc, end - pc);
+    Span<const unsigned char> data;
+    bool ret = GetScriptOp(script, opcodeRet, &data);
+    if (pvchRet) *pvchRet = std::vector<unsigned char>(data.begin(), data.end());
```

Comment:
This could also be done without constructing a new vector instance through `pvchRet->assign(...,...);`?

Reply:
Good idea, done.